rfc/rfc-0006-child-resource-names.md

```diff
@@ -0,0 +1,43 @@
+# RFC-0006: Child Resource Names
+
+**Authors:** Scott Andrews
+
+**Status:**
+
+**Pull Request URL:** [#1369](https://github.com/projectriff/riff/pull/1369)
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** N/A
+
+
+## Problem
+When a controller creates a new resource, it need to either pick a static name, or let kubernetes generate a name. Static names are predictable, but may already be in use. Generated names cannot be predicated, but are guaranteed to be available.
+
+### Anti-Goals
+This RFC does not prescribe which naming scheme to use for any specific resource.
+
+## Solution
+While there will not be a cut and dry answer for each situation, there are heuristics to guide the decision:
+- if a resource is not consumed directly by a non-controlled resource by name, generate the name (e.g. Pods in a ReplicaSet)
```

Comment:
```suggestion
- if a resource is not consumed directly by a non-controlling resource by name, generate the name (e.g. Pods in a ReplicaSet)
```

Reply:
I'm not sure what this is getting at. It hints at various possibilities:
* a resource which is consumed directly by name or not (consumed indirectly or not by name or both or not consumed at all)
* the type of consumer ("controlled resource" which might mean created by a controller, "controlling resource" which might mean the consuming resource causes a controller to generate the consumed resource, something other than a resource such as a user)

and it only stipulates generating the name in one specific combination of possibilities.